temporal-sdk/src/main/java/io/temporal/failure/ApplicationFailure.java

```diff
@@ -21,6 +21,7 @@
 package io.temporal.failure;
 
 import com.google.common.base.Strings;
+import io.temporal.api.enums.v1.ApplicationErrorCategory;
```

Comment:
We shouldn't use the proto enum in the user facing API, we should declare our own enum in the Java SDK and convert.

Reply:
sorry I didn't notice before you were still using the proto enum

Reply:
I formerly had this but: https://github.com/temporalio/sdk-java/pull/2485#discussion_r2054380000

It looked like we use raw protos, when do we use each pattern?

Reply:
All new features should NOT use raw proto, only old features use raw protos